src/components/TextButton/TextButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+import { Text, TouchableWithoutFeedback, View } from "react-native";
+import { Colors } from "@constants/constants";
+import styles from "./styles";
+
+const TextButton = ({ text, addOns, isDisabled = false, styles: customStyles, onPress }: TextButtonProps) => {
+  const [isPressIn, setIsPressIn] = useState(false);
+
+  const handlePressIn = () => setIsPressIn(true);
+
+  const handlePressOut = () => setIsPressIn(false);
+
+  return (
+    <View style={styles.outsideView}>
+      <TouchableWithoutFeedback
+        onPress={isDisabled ? undefined : onPress}
+        onPressIn={isDisabled ? undefined : handlePressIn}
```

Comment:
why define isDisabled for onPress seperately when you have already used disabled property